Android/Java/app/src/main/java/com/ironsource/ironsourcesdkdemo/DemoBannerAdListener.java

```diff
@@ -17,59 +17,82 @@ public DemoBannerAdListener(DemoActivityListener listener) {
         this.listener.setBannerViewVisibility(View.VISIBLE);
         this.listener.setEnablementForButton(DemoButtonIdentifiers.LOAD_BANNER_BUTTON_IDENTIFIER, false);
         this.listener.setEnablementForButton(DemoButtonIdentifiers.DESTROY_BANNER_BUTTON_IDENTIFIER, true);
     }
 
     /**
-     Called after a banner has attempted to load an ad but failed.
-     This delegate will be sent both for manual load and refreshed banner failures.
-     @param ironSourceError The reason for the error.
+     Called after a banner has attempted to load an ad but failed
+     This delegate will be sent both for manual load and refreshed banner failures
+     @param error The reason for the error
+     */
+    @Override
+    public void onAdLoadFailed(@NonNull LevelPlayAdError error) {
+        logCallbackName(TAG, "error = " + error);
+    }
+
+    /**
+     Called after a banner was displayed and visible on screen
+     @param adInfo The info of the ad
+     */
+    @Override
+    public void onAdDisplayed(@NonNull LevelPlayAdInfo adInfo) {
+        logCallbackName(TAG, "adInfo = " + adInfo);
+    }
+
+    /**
+     Called after a banner failed to be displayed on screen
+     @param adInfo The info of the ad
+     @param error The reason for the error
      */
+
     @Override
-    public void onAdLoadFailed(IronSourceError ironSourceError) {
-        logCallbackName(TAG, "error = " + ironSourceError);
+    public void onAdDisplayFailed(@NonNull LevelPlayAdInfo adInfo, @NonNull LevelPlayAdError error) {
+        logCallbackName(TAG, "error = " + error + " | adInfo = " + adInfo);
+
     }
 
     /**
-     Called after a banner has been clicked.
-     @param adInfo The info of the ad.
+     Called after a banner has been clicked
+     @param adInfo The info of the ad
      */
     @Override
-    public void onAdClicked(AdInfo adInfo) {
+    public void onAdClicked(@NonNull LevelPlayAdInfo adInfo) {
         logCallbackName(TAG, "adInfo = " + adInfo);
     }
 
     /**
-     Called when a user was taken out of the application context.
-     @param adInfo The info of the ad.
+     Called when a banner opened on full screen
+     @param adInfo The info of the ad
      */
     @Override
-    public void onAdLeftApplication(AdInfo adInfo) {
+    public void onAdExpanded(@NonNull LevelPlayAdInfo adInfo) {
         logCallbackName(TAG, "adInfo = " + adInfo);
     }
 
     /**
-     Called when a banner presented a full screen content.
-     @param adInfo The info of the ad.
+     Called after a banner is restored to its original size
+     @param adInfo The info of the ad
      */
     @Override
-    public void onAdScreenPresented(AdInfo adInfo) {
+    public void onAdCollapsed(@NonNull LevelPlayAdInfo adInfo) {
         logCallbackName(TAG, "adInfo = " + adInfo);
     }
 
     /**
-     Called after a full screen content has been dismissed.
-     @param adInfo The info of the ad.
+     Called when a user pressed on the ad and was navigated out of the app
```

Comment:
This callback is called anyway when the app is minimized? not only when the user click on the ad.

Reply:
this text is aligned to the knowledge center. I would not reccommed writing something different, unless the text there is wrong. both texts are correct
https://developers.is.com/ironsource-mobile/android/banner-integration-sdk-8-1-0-alpha-with-adaptive/#step-3